### ISSUE TITLE
Detect missing NO-ERROR on FIND ... IF AVAILABLE

### ISSUE THREAD

**Issue by lievendf** [Riverside-Software/sonar-openedge#1108]
When writing
```
FIND Order WHERE ... NO-LOCK.
IF AVAILABLE Order
THEN ...
```
Sonar should warn for a missing NO-ERROR, as the IF AVAILABLE will not be executed in case the record isn't found.

**Comment by gquerret** [comment]
Did you enable this rule ? https://sonar.riverside-software.fr/coding_rules?languages=oe&q=error&open=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.FindNoError

**Comment by cdelabriere** [comment]
Hello ! I close this issue, the rule "FindNoError" should solve your case. 
Feel free to update this issue if not.

**Comment by lievendf** [comment]
Hi @cdelabriere , I believe the rule FindNoError does not apply here.
I explicitly **don't** want to use NO-ERROR in cases where I'm sure the record must exist, and therefore I would **want** to get an error if it isn't found.
In this case I would like Sonar to warn me for the use of IF AVAILABLE, as that makes no sense there: either the record is found and the code continues, or it isn't and an error is raised and the next statement is not executed at all...
Can this issue be reopened?

**Comment by gquerret** [comment]
Rule implemented, will be available in next version
